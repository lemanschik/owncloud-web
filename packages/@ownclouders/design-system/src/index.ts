import { getSizeClass } from './utils/sizeClasses'
import './utils/webFontLoader'

import * as components from './components'
import * as directives from './directives'

const initializeCustomProps = (tokens = [], prefix) => {
  for (const param in tokens) {
    initializeCustomProp(prefix + param, tokens[param])
  }
}

const initializeCustomProp = (key: string, value: string | undefined) => {
  if (value === undefined) {
    return
  }
  ;(document.querySelector(':root') as HTMLElement).style.setProperty('--oc-' + key, value)
}

export default {
  install(app, options: any = {}) {
    const themeOptions = options.tokens
    initializeCustomProps(themeOptions?.breakpoints, 'breakpoint-')
    initializeCustomProps(themeOptions?.colorPalette, 'color-')
    initializeCustomProps(themeOptions?.fontSizes, 'font-size-')
    initializeCustomProps(themeOptions?.sizes, 'size-')
    initializeCustomProps(themeOptions?.spacing, 'space-')
    initializeCustomProp('font-family', themeOptions?.fontFamily)

    Object.values(components).forEach((c) => app.component(c.name, c))
    Object.values(directives).forEach((d) => app.directive(d.name, d))
  }
}

export const utils = { getSizeClass }
export * as components from './components'
export * as composables from './composables'
export * as directives from './directives'
export * as helpers from './helpers'
