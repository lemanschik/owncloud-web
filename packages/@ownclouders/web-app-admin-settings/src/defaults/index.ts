export const supportedLogoMimeTypes = ['image/gif', 'image/jpeg', 'image/png']
