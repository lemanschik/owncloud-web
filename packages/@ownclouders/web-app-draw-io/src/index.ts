import translations from '../l10n/translations.json'
import App from './App.vue'

const routes = [
  {
    name: 'draw-io',
    path: '/:driveAliasAndItem(.*)?',
    component: App,
    meta: {
      authContext: 'hybrid',
      patchCleanPath: true
    }
  }
]

const appInfo = {
  name: 'Draw.io',
  id: 'draw-io',
  icon: 'grid',
  extensions: [
    {
      extension: 'drawio',
      routeName: 'draw-io',
      newFileMenu: {
        menuTitle($gettext) {
          return $gettext('Draw.io document')
        }
      }
    },
    {
      extension: 'vsdx',
      routeName: 'draw-io'
    }
  ]
}

export default {
  appInfo,
  routes,
  translations
}
