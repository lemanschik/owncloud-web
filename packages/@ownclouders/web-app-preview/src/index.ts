import translations from '../l10n/translations.json'
import * as app from './App.vue'
const { default: App, mimeTypes, appId } = app as any

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const routes = [
  {
    path: '/:driveAliasAndItem(.*)?',
    component: App,
    name: 'media',
    meta: {
      authContext: 'hybrid',
      title: $gettext('Preview'),
      patchCleanPath: true
    }
  }
]

const routeName = 'preview-media'

const appInfo = {
  name: $gettext('Preview'),
  id: appId,
  icon: 'eye',
  extensions: mimeTypes().map((mimeType) => ({
    canBeDefault: true,
    mimeType,
    routeName,
    label: $gettext('Preview')
  }))
}

export default {
  appInfo,
  routes,
  translations,
  mimeTypes
}
