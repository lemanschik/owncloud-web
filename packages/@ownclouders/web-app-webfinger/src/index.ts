import translations from '../l10n/translations.json'
import Resolve from './views/Resolve.vue'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Webfinger'),
  id: 'webfinger',
  icon: 'fingerprint',
  isFileEditor: false
}

const routes = () => [
  {
    name: 'webfinger-root',
    path: '/',
    redirect: () => {
      return { name: 'webfinger-resolve' }
    },
    meta: {
      authContext: 'anonymous'
    }
  },
  {
    path: '/resolve',
    name: 'webfinger-resolve',
    component: Resolve,
    meta: {
      authContext: 'idp',
      title: $gettext('Resolve destination'),
      entryPoint: true
    }
  }
]

export default {
  appInfo,
  routes,
  translations
}
