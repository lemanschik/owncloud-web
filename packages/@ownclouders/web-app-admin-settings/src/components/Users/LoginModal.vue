<template>
  <oc-modal
    :title="modalTitle"
    :button-cancel-text="$gettext('Cancel')"
    :button-confirm-text="$gettext('Confirm')"
    :button-confirm-disabled="!selectedOption"
    @cancel="$emit('cancel')"
    @confirm="$emit('confirm', { users, value: selectedOption.value })"
  >
    <template #content>
      <oc-select
        :model-value="selectedOption"
        :label="$gettext('Login')"
        :options="options"
        :placeholder="$gettext('Select...')"
        :warning-message="
          currentUserSelected ? $gettext('Your own login status will remain unchanged.') : ''
        "
        @update:model-value="changeSelectedOption"
      />
    </template>
  </oc-modal>
</template>

<script lang="ts">
import { computed, defineComponent, onMounted, PropType, ref, unref } from 'vue'
import { useGettext } from 'vue3-gettext'
import { User } from 'web-client/src/generated'
import { useStore } from 'web-pkg/src'

type LoginOption = {
  label: string
  value: boolean
}

export default defineComponent({
  name: 'LoginModal',
  props: {
    users: {
      type: Array as PropType<User[]>,
      required: true
    }
  },
  emits: ['confirm', 'cancel'],
  setup(props) {
    const store = useStore()
    const { $gettext, $ngettext } = useGettext()

    const selectedOption = ref()
    const options = ref([
      { label: $gettext('Allowed'), value: true },
      { label: $gettext('Forbidden'), value: false }
    ])

    const changeSelectedOption = (option: LoginOption) => {
      selectedOption.value = option
    }

    const currentUserSelected = computed(() => {
      return props.users.some((u) => u.id === store.getters.user.uuid)
    })

    const modalTitle = computed(() => {
      return $ngettext(
        'Edit login for "%{user}"',
        'Edit login for %{userCount} users',
        props.users.length,
        {
          user: props.users[0].displayName,
          userCount: props.users.length.toString()
        }
      )
    })

    onMounted(() => {
      if (props.users.every((u) => u.accountEnabled !== false)) {
        selectedOption.value = unref(options).find(({ value }) => !!value)
      } else if (props.users.every((u) => u.accountEnabled === false)) {
        selectedOption.value = unref(options).find(({ value }) => !value)
      }
    })

    return {
      selectedOption,
      options,
      changeSelectedOption,
      currentUserSelected,
      modalTitle
    }
  }
})
</script>
