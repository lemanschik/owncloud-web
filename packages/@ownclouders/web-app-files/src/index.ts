import App from './App.vue'
import Favorites from './views/Favorites.vue'
import FilesDrop from './views/FilesDrop.vue'
import SharedWithMe from './views/shares/SharedWithMe.vue'
import SharedWithOthers from './views/shares/SharedWithOthers.vue'
import SharedViaLink from './views/shares/SharedViaLink.vue'
import SpaceDriveResolver from './views/spaces/DriveResolver.vue'
import SpaceProjects from './views/spaces/Projects.vue'
import TrashOverview from './views/trash/Overview.vue'
import translations from '../l10n/translations.json'
import quickActions from './quickActions'
import store from './store'
import { SDKSearch } from './search'
import { eventBus } from 'web-pkg/src/services/eventBus'
import { Registry, ArchiverService } from './services'
import fileSideBars from './fileSideBars'
import { buildRoutes } from './router'
import get from 'lodash-es/get'
import { AppNavigationItem, AppReadyHookArgs } from 'web-pkg/src/apps'

// dirty: importing view from other extension within project
import SearchResults from '../../web-app-search/src/views/List.vue'
import { isPersonalSpaceResource } from 'web-client/src/helpers'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Files'),
  id: 'files',
  icon: 'resource-type-folder',
  isFileEditor: false,
  extensions: [],
  fileSideBars
}
const navItems = (context): AppNavigationItem[] => {
  return [
    {
      name(capabilities) {
        return capabilities.spaces?.enabled ? $gettext('Personal') : $gettext('All files')
      },
      icon: appInfo.icon,
      route: {
        path: `/${appInfo.id}/spaces/personal`
      },
      enabled(capabilities) {
        if (!capabilities.spaces?.enabled) {
          return true
        }
        return !!context?.$store?.getters['runtime/spaces/spaces'].find((drive) =>
          isPersonalSpaceResource(drive)
        )
      }
    },
    {
      name: $gettext('Favorites'),
      icon: 'star',
      route: {
        path: `/${appInfo.id}/favorites`
      },
      enabled(capabilities) {
        return capabilities.files?.favorites
      }
    },
    {
      name: $gettext('Shares'),
      icon: 'share-forward',
      route: {
        path: `/${appInfo.id}/shares`
      },
      activeFor: [{ path: `/${appInfo.id}/spaces/share` }],
      enabled(capabilities) {
        return capabilities.files_sharing?.api_enabled !== false
      }
    },
    {
      name: $gettext('Spaces'),
      icon: 'layout-grid',
      route: {
        path: `/${appInfo.id}/spaces/projects`
      },
      activeFor: [{ path: `/${appInfo.id}/spaces/project` }],
      enabled(capabilities) {
        return capabilities.spaces?.projects
      }
    },
    {
      name: $gettext('Deleted files'),
      icon: 'delete-bin-5',
      route: {
        path: `/${appInfo.id}/trash/overview`
      },
      activeFor: [{ path: `/${appInfo.id}/trash` }],
      enabled(capabilities) {
        return capabilities.dav?.trashbin === '1.0'
      }
    }
  ]
}

export default {
  appInfo,
  store,
  routes: buildRoutes({
    App,
    Favorites,
    FilesDrop,
    SearchResults,
    Shares: {
      SharedViaLink,
      SharedWithMe,
      SharedWithOthers
    },
    Spaces: {
      DriveResolver: SpaceDriveResolver,
      Projects: SpaceProjects
    },
    Trash: {
      Overview: TrashOverview
    }
  }),
  navItems,
  quickActions,
  translations,
  ready({ router, store, globalProperties }: AppReadyHookArgs) {
    const { $clientService } = globalProperties
    Registry.sdkSearch = new SDKSearch(store, router, $clientService)

    // when discussing the boot process of applications we need to implement a
    // registry that does not rely on call order, aka first register "on" and only after emit.
    eventBus.publish('app.search.register.provider', Registry.sdkSearch)
    globalProperties.$archiverService = new ArchiverService(
      $clientService,
      store.getters.configuration.server || window.location.origin,
      get(store, 'getters.capabilities.files.archivers', [
        {
          enabled: true,
          version: '1.0.0',
          formats: ['tar', 'zip'],
          archiver_url: `${store.getters.configuration.server}index.php/apps/files/ajax/download.php`
        }
      ])
    )
  }
}
