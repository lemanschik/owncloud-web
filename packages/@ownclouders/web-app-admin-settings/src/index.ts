import translations from '../l10n/translations.json'
import General from './views/General.vue'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
import { Ability } from 'web-pkg'
import { AppNavigationItem } from 'web-pkg/src/apps'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

const routes = ({ $ability }: { $ability: Ability }) => [
  {
    path: '/',
    redirect: () => {
      if ($ability.can('read-all', 'Setting')) {
        return { name: 'admin-settings-general' }
      }
      if ($ability.can('read-all', 'Account')) {
        return { name: 'admin-settings-users' }
      }
      if ($ability.can('read-all', 'Group')) {
        return { name: 'admin-settings-groups' }
      }
      if ($ability.can('read-all', 'Space')) {
        return { name: 'admin-settings-spaces' }
      }
      throw Error('Insufficient permissions')
    }
  },
  {
    path: '/general',
    name: 'admin-settings-general',
    component: General,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Setting')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('General')
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Account')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Group')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Space')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = ({ $ability }: { $ability: Ability }): AppNavigationItem[] => [
  {
    name: $gettext('General'),
    icon: 'settings-4',
    route: {
      path: `/${appInfo.id}/general?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Setting')
    }
  },
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Account')
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Group')
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Space')
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
