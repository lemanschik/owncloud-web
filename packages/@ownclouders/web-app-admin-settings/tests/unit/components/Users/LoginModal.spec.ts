import LoginModal from '../../../../src/components/Users/LoginModal.vue'
import {
  createStore,
  defaultPlugins,
  defaultStoreMockOptions,
  shallowMount
} from 'web-test-helpers'
import { mock } from 'jest-mock-extended'
import { User } from 'web-client/src/generated'

describe('LoginModal', () => {
  it('renders the input including two options', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('shows a warning when the current user is being selected', () => {
    const { wrapper } = getWrapper([mock<User>({ id: '1' })])
    expect(wrapper.findComponent<any>('oc-select-stub').props('warningMessage')).toBeDefined()
  })
})

function getWrapper(users = [mock<User>()]) {
  const storeOptions = defaultStoreMockOptions
  storeOptions.getters.user.mockReturnValue({ uuid: '1' })
  const store = createStore(storeOptions)
  return {
    wrapper: shallowMount(LoginModal, {
      props: {
        users
      },
      global: {
        renderStubDefaultSlot: true,
        plugins: [...defaultPlugins(), store],
        stubs: { OcModal: false }
      }
    })
  }
}
