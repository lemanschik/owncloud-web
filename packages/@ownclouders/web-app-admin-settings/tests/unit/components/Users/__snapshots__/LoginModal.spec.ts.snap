// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal renders the input including two options 1`] = `
<div aria-labelledby="oc-modal-title" class="oc-modal-background">
  <focus-trap-stub active="true" allowoutsideclick="true" clickoutsidedeactivates="false" delayinitialfocus="true" escapedeactivates="true" initialfocus="[Function]" preventscroll="false" returnfocusondeactivate="true" setreturnfocus="false">
    <div aria-modal="true" class="oc-modal oc-modal-passive" role="dialog" tabindex="0">
      <div class="oc-modal-title">
        <!--v-if-->
        <h2 id="oc-modal-title">Edit login for "function () {
          return fn.apply(this, arguments);
          }"</h2>
      </div>
      <div class="oc-modal-body">
        <div class="oc-modal-body-message">
          <oc-select-stub clearable="false" disabled="false" filter="[Function]" fixmessageline="false" id="oc-select-1" label="Login" loading="false" multiple="false" optionlabel="label" options="[object Object],[object Object]" placeholder="Select..." readonly="false" searchable="true" warningmessage=""></oc-select-stub>
        </div>
        <!--v-if-->
      </div>
      <div class="oc-modal-body-actions oc-flex oc-flex-right">
        <oc-button-stub appearance="outline" class="oc-modal-body-actions-cancel" disabled="false" gapsize="medium" justifycontent="center" size="medium" submit="button" type="button" variation="passive">Cancel</oc-button-stub>
        <!--v-if-->
        <oc-button-stub appearance="filled" class="oc-modal-body-actions-confirm oc-ml-s" disabled="true" gapsize="medium" justifycontent="center" size="medium" submit="button" type="button" variation="primary">Confirm</oc-button-stub>
      </div>
    </div>
  </focus-trap-stub>
</div>
`;
